maliput_sim/export/base.py

```diff
@@ -0,0 +1,41 @@
+# BSD 3-Clause License
+#
+# Copyright (c) 2023, Woven by Toyota
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from
+#    this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from abc import ABC, abstractmethod
+
+class ExportBase(ABC):
+    """Base class for export."""
+
+    def __init__(self, simulation):
+        self._simulation = simulation
+
+    @abstractmethod
+    def dump(config):
```

Comment:
I have. Having a functor here makes me think if this should be a simply method

Reply:
It depends on how you are planning to use this functionality.

As it is presented now, yes, you are right. You should consider making it a function. However, there is room for further sophistication in the export interface.

Typically, you would like to consider adding cursors for time filter, topic filters, etc etc. All those configurations can be abstracted in the config dictionary but then you need to parse/look up those into individual elements when you may be better off with individual methods that do different export types.

I suggest keeping the interface for now, and replacing the dictionary for specific named arguments, and finally using an `extra_config` which is implementation dependent if required. Note that most probably the `extra_config` argument is unlikely needed when dumping the simulation because you also have the constructors which gives you the 
advantage of passing implementation specific information. What about the following?

```python
class Exporter(ABC):
    def __init__(self, simulation: Simulation):
      self._simulation = simulation

    @abstractmethod
    def dump(self, export_agent_param: bool = True, export_road_network_param: bool = True, init_timestamp: float = None, end_timestamp: float = None, export_signals : list = None):
        pass
``` 

And then MCAPExporter offers:

```
class MCAPExporter(Exporter):
    def __init__(self, simulation: Simulation, output_filepath: str =None):
      self._output_filepath = output_filepath
      super().__init__(simulation)
...
```

Reply:
I like the proposal.

> init_timestamp: float = None, end_timestamp: float = None,

What will be the use case? Regarding the init_timestamp you might want to set up a init timestamp and then the sim time simply goes as addition of that init timestamp. And what about the end_timestamp?

> export_signals : list = None

Is this related to the controllers' state?

Reply:
> What will be the use case? Regarding the init_timestamp you might want to set up a init timestamp and then the sim time simply goes as addition of that init timestamp. And what about the end_timestamp?

These are just two time cursors to select the section of the state progression you care about.

> Is this related to the controllers' state?

And components value progression as well.

Reply:
> These are just two time cursors to select the section of the state progression you care about.

Like for indicating " I want to export the chunk from init: 3.2 sec to end 6.1 sec"
Ok, I thought that the `init time` was to indicate some kind of offset (but just value, not omiting any data). Your case is more interesting.

I will address it. it is still pending.

Reply:
Addressed!